extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -185,24 +192,46 @@ public QuartzScheduler(SchedulerContext context, QuartzSupport quartzSupport, Co
 
     @Override
     public void pause() {
-        try {
-            if (scheduler != null) {
-                scheduler.pauseAll();
+        if (!enabled) {
+            LOGGER.warn("Quartz Scheduler is disabled and cannot be paused");
+        } else {
+            try {
+                if (scheduler != null) {
+                    scheduler.standby();
+                }
+            } catch (SchedulerException e) {
+                LOGGER.warn("Unable to pause scheduler", e);
             }
-        } catch (SchedulerException e) {
-            LOGGER.warn("Unable to pause scheduler", e);
         }
     }
 
     @Override
     public void resume() {
-        try {
-            if (scheduler != null) {
-                scheduler.resumeAll();
+        if (!enabled) {
+            LOGGER.warn("Quartz Scheduler is disabled and cannot be resumed");
+        } else {
+            try {
+                if (scheduler != null) {
+                    scheduler.start();
+                }
+            } catch (SchedulerException e) {
+                LOGGER.warn("Unable to resume scheduler", e);
+            }
+        }
+    }
+
+    @Override
+    public boolean isRunning() {
+        if (!enabled || scheduler == null) {
+            return false;
+        } else {
+            try {
+                return !scheduler.isInStandbyMode();
+            } catch (SchedulerException e) {
+                LOGGER.warn("Unable to resume scheduler", e);
```

Comment:
Message? In fact, I'd just throw an `IllegalStateException` instead of a warning. Although it seems that `isInStandbyMode()` never throws an expception...